python/orion_finance_sdk_py/contracts.py

```
@@ -163,15 +176,74 @@ def __init__(self, contract_name: str, contract_address: str):
+        """Resolve a datetime, unix timestamp, or block number to a block number."""
+        if isinstance(value, datetime):
+            ts = (
+                value
+                if value.tzinfo is not None
+                else value.replace(tzinfo=timezone.utc)
+            )
+            return self.block_at_timestamp(int(ts.timestamp()))
+        if isinstance(value, int):
+            if value >= _TIMESTAMP_THRESHOLD:
+                return self.block_at_timestamp(value)
+            return value
```

Ruff (0.15.21)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)